geopandas/tests/test_crs.py

```diff
@@ -282,6 +282,13 @@ def test_dataframe(self):
         df.crs = 27700
         assert df.crs == self.osgb
 
+        df = GeoDataFrame()
+        df.crs = 4326
+        df["geometry"] = None
```

Comment:
This reminds me, we should probably also support `df['geometry'] = scalar_shapely_geom`. So maybe we shouldn't special case `None`, but rather handle scalars (non list-likes) in general